### ISSUE TITLE
Pst

### ISSUE THREAD

**Issue by wyzhangMPI** [severinEvo/gene_expression#3]
Hi,
   It is great to read your ME paper, and find the code to calcualte the gene expression differentiation between two groups.
   I am wondering if it is possible to apply this function to the conditions with more than 2 populations, say 3?
   Or do you known there is other approahes to do this way?
   Thanks!

Best,
Wenyu Zhang

**Comment by severinEvo** [comment]
Thank you for your interest in our work! Running the method with more than two groups is no problem, you can just specify that in the code. Let's say you have four samples for each group, and three populations, two sexes and two organs. You would run the code like this:

```
population <- rep(c("A","B","C"),each=8,times=2) # 8 samples per population (4 female, 4 male) * 2 organs
organ <- rep(c("organ1","organ2"),each=24) # 24 samples per organ: 4 per sex * 3 populations
sex <- rep(c("f","m"),each=4,times=3*2)

pst_data <- apply(dataset, 1, Pst, fixed=population, organ=organ, sex=sex)
```
Make sure to use your populations as the fixed factor and add as many other factors as you have in your design. So if all samples are from the same sex or organ, simply leave that factor out.

I hope that helps! Let me know if you run into any other issues.

**Comment by wyzhangMPI** [comment]
Hi，
     Thanks for the kind response. Now it works.
     However, I still have other related questions: 1) Are the Pst values 
normalized and directly comparable for different genes, i.e., reflecting 
the extent of contribution to population differentiation? 
2)Occasionally, there are some genes showing negative Pst values, does 
them mean that they play opposite role in population differentiation or 
basically mean they are not relevant?

     Thanks again!

------ Original Message ------

**Comment by severinEvo** [comment]
Hi, glad I could help.

1) I would be cautious when comparing *P*<sub>ST</sub> between genes. I'm not saying that it is a bad idea, but just keep in mind what a high or low *P*<sub>ST</sub> value means. A high *P*<sub>ST</sub> value indicates that the between-population variance (*v*<sub>B</sub>) is larger than the within-population variance (*v*<sub>W</sub>), meaning that it could be due to low *v*<sub>W</sub> or due to high *v*<sub>B</sub> or both. It is a good idea to check. So to get back to your question: *P*<sub>ST</sub> values are *v*<sub>B</sub> normalized by *v*<sub>W</sub>. So if this is the measure you want to compare genes with each other, then that's great! I'd suggest that beside looking at *P*<sub>ST</sub>, you should also look at the two variance components to understand their distributions, their extremes, and their respective effects on the *P*<sub>ST</sub> values.

2) The method does produce some amount of negative values for *P*<sub>ST</sub>, *v*<sub>W</sub> and *v*<sub>B</sub>. Some people like to bound *P*<sub>ST</sub> at 0, and you can do so simply by setting all negative values to 0. However, I like to have negative values in there, because the distributions are better for forming correlations. You may choose depending on what your goal is.

I'm closing this issue now as it seems we have figured out your problem. If you have further questions on running the method or interpreting the results, feel free to email me at severin.uebbing-ät'yale.edu
